odl/solvers/functional/default_functionals.py

```diff
@@ -464,3 +464,67 @@ def __init__(self, space):
             Domain of the functional.
         """
         super().__init__(space=space, constant=0)
+
+
+class IndicatorBox(Functional):
+
+    """Indicator on some box shaped domain."""
+
+    def __init__(self, space, lower, upper):
+        """Initialize an instance
+
+        Parameters
+        ----------
+        space : `LinearSpace`
+            Domain of the functional G(x)
+        lower : ``space.field`` element or ``space`` `element-like`, optional
+            The lower bound.
+            Default: ``None``, interpreted as -infinity
+        upper : ``space.field`` element or ``space`` `element-like`, optional
+            The upper bound.
+            Default: ``None``, interpreted as +infinity
+
+        Examples
+        --------
+        >>> import odl
+        >>> space = odl.rn(3)
+        >>> func = IndicatorBox(space, 0, 2)
+        >>> func([0, 1, 2])  # all points inside
+        0
+        >>> func([0, 1, 3])  # one point outside
+        inf
+        """
+        Functional.__init__(self, space, linear=False)
+        self.lower = lower
+        self.upper = upper
+
+    def _call(self, x):
+        """Apply the functional to the given point."""
+        out = self.domain.element()
```

Comment:
Explain what magic happens here, also perhaps optimize for the case where the bounds are scalar.